src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -292,11 +297,16 @@ public void findByBirthdateBetweenAndUsernameContainingEmpty() throws Exception
         LocalDate fromBirthdate = mockUser.getBirthdate().plusMonths(2);
         LocalDate toBirthdate = mockUser.getBirthdate().plusMonths(8);
         String username = mockUser.getUsername();
+
+        List<User> allUsers = Arrays.asList(mockUser);
+        PageRequest pageRequest = new PageRequest(0,2);
+
+        given(serviceUser.findByBirthdateBetweenAndUsernameContainingIgnoreCase(fromBirthdate, toBirthdate, username, pageRequest)).willReturn(new PageImpl<>(allUsers, pageRequest, 2));
```

Comment:
The issue here is that you're setting in the mock a page request of 0,2 but you're not putting that as query params in the url so that the Pageable that gets built in the controller, has those parameters, and defaults it to a pageable of 0,20. Thus, the mock will never work.
I would do this using any(PageRequest.class) and also matchers on the other parameters just as I commented in the bookcontroller test.